go.mod

```diff
@@ -11,24 +11,24 @@ require (
 	github.com/gophercloud/gophercloud v1.3.0
 	github.com/gophercloud/utils v0.0.0-20221207145018-e8fba78967ca
 	github.com/hashicorp/go-version v1.4.0
-	github.com/onsi/ginkgo/v2 v2.9.2
-	github.com/onsi/gomega v1.27.6
+	github.com/onsi/ginkgo/v2 v2.9.5
+	github.com/onsi/gomega v1.27.7
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/client_golang v1.15.1
 	github.com/spf13/pflag v1.0.5
 	golang.org/x/crypto v0.7.0
-	golang.org/x/text v0.8.0
+	golang.org/x/text v0.9.0
 	gopkg.in/ini.v1 v1.67.0
-	k8s.io/api v0.26.3
-	k8s.io/apiextensions-apiserver v0.26.3
-	k8s.io/apimachinery v0.26.3
-	k8s.io/client-go v0.26.3
-	k8s.io/component-base v0.26.3
-	k8s.io/klog/v2 v2.90.1
-	k8s.io/utils v0.0.0-20230313181309-38a27ef9d749
-	sigs.k8s.io/cluster-api v1.4.1
-	sigs.k8s.io/cluster-api/test v1.4.1
-	sigs.k8s.io/controller-runtime v0.14.6
+	k8s.io/api v0.27.2
+	k8s.io/apiextensions-apiserver v0.27.2
+	k8s.io/apimachinery v0.28.0-alpha.0.0.20230524221503-b43878988803
+	k8s.io/client-go v0.27.2
+	k8s.io/component-base v0.27.2
+	k8s.io/klog/v2 v2.100.1
+	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2
+	sigs.k8s.io/cluster-api v1.4.0-beta.2.0.20230526073652-9b1ead8d49ee
+	sigs.k8s.io/cluster-api/test v1.4.0-beta.2.0.20230526073652-9b1ead8d49ee
```

Comment:
These all seem to have gone backwards.

Reply:
This should be IMO ok - this is just the main branch of `cluster-api`; `sigs.k8s.io/cluster-api@main`

Reply:
Yeah the "version" looks weird because of how go resolves them. It looks for the latest tag in the git history. So since the later tags (e.g. v1.4.2) is on a release branch, it will not find it. Instead it finds the latest tag that sits in the main branch.

That said, I'm not sure we should pull in main here. It makes more sense to stick to actual releases.

Reply:
This is needed to be done because the newest cluster-api version is v1.4.2 which doesn't contain the bumps to 1.27 (including big changes in controller-runtime package).
The release including this change might take some time as well.

Reply:
Why not then wait for it to land?